src/metabase/query_processor/middleware/optimize_temporal_filters.clj

```diff
@@ -99,13 +154,16 @@
       (_ :guard (partial mbql.u/is-clause? (set (keys (methods optimize-filter)))))
       (if (can-optimize-filter? &match)
         (let [optimized (optimize-filter &match)]
+          (when-not optimized
+            (throw (ex-info (tru "Error optimizing temporal filter clause")
+                            {:clause &match})))
           (when-not (= &match optimized)
             (log/tracef "Optimized filter %s to %s" (pr-str &match) (pr-str optimized)))
           optimized)
         &match))))
 
 (defn optimize-temporal-filters
-  "Middlware that optimizes equality (`=` and `!=`) and comparison (`<`, `between`, etc.) filter clauses against
+  "Middlware that optimizes equality (`=` and `!=`) and comparison (`<`, `between`, etc.) filter clauses agat
```

Comment:
`against` -> `agat` looks accidental

Reply:
Totally is. Oops. Not sure how I did that